### PR TITLE
 tests: add interactive_sync adapted to shell 

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR changes the way Interactive Sync is done when the shell is present, instead of using new commands it prompts the shell with a `\n` until `>` is received as an answer

This will reduce the size of a lot of test applications. 

### Testing procedure

- Run all tests using shell on multiple boards (I'll run it on my setup as soon as the PR is open)

- Green murdock

- `tests/periph_wdt` and `tests/struc_tm_utility/` should still work:

```
PASSED target time: 128[ms], actual_time: 128.534[ms]
PASSED target time: 512[ms], actual_time: 502.025[ms]
PASSED target time: 1024[ms], actual_time: 1002.731[ms]
PASSED target time: 8192[ms], actual_time: 8001.558[ms]
TEST PASSED
```

```
BOARD=samr21-xpro make -C tests/struct_tm_utility/ flash test

>  day 2018 11 28
What weekday was 2018-11-28? The 332(th) day of the year was a WED.
day 2016 2 29
>  day 2016 2 29
What weekday was 2016-02-29? The 60(th) day of the year was a MON.
day 2017 2 29
>  day 2017 2 29
The supplied date is invalid, but no error should occur.
```

- tested with `miniterm`, `socat` and `pyterm`, can someone test for `rtt`?

### Issues/PRs references

Fixes compilation issues in #13152

### CODE REVIEW

**Comment by fjmolinas** [comment]
triggering build since there is no queue

File: dist/pythonlibs/testrunner/utils.py [hunk 1]
```diff
@@ -20,8 +20,24 @@ def test_utils_interactive_sync(child, retries, delay):
...
     child.expect_exact('START')
+
+
+def test_utils_interactive_sync_shell(child, retries, delay):
+    """Synchronisation for 'test_utils_interactive_sync' function
+    when shell module is used.
+
+    Interacts through input to wait for node being ready.
+    """
+    for _ in range(0, retries):
```
Comment (kaspar030) [inline dist/pythonlibs/testrunner/utils.py]:
```suggestion
    for _ in range(retries):
```

File: dist/pythonlibs/testrunner/utils.py [hunk 2]
```diff
@@ -20,8 +20,24 @@ def test_utils_interactive_sync(child, retries, delay):
...
+        child.sendline('\n')
+        ret = child.expect_exact(['>', pexpect.TIMEOUT], timeout=delay)
+        if ret == 0:
+            break
+    else:
+        # Last call to make it fail here,
+        child.expect_exact('>', timeout=0)
```
Comment (miri64) [inline dist/pythonlibs/testrunner/utils.py]: Both `test_utils_interactive_sync()` and `test_utils_interactive_sync_shell()` have the same code and just use some different values. How about generalizing the two? 

```py
def test_utils_interactive_sync(child, retries, delay, ready_cmd='r', ready_exp='READY'):
    """Synchronisation for 'test_utils_interactive_sync' function
    when shell module is used.
    Interacts through input to wait for node being ready.
    """
    for _ in range(0, retries):
        child.sendline(ready_cmd)
        ret = child.expect_exact([ready_exp, pexpect.TIMEOUT], timeout=delay)
        if ret == 0:
            break
    else:
        # Last call to make it fail here,
        child.expect_exact(ready_exp, timeout=0)


def test_utils_interactive_sync_shell(child, retries, delay):
    test_utils_interactive_sync(child, retries, delay, '\n', '>')
```

File: dist/pythonlibs/testrunner/utils.py (hunk 1 above)
Comment (miri64) [reply dist/pythonlibs/testrunner/utils.py]: (exists in `test_utils_interactive_sync()` as well and should be piggybacked)

File: dist/pythonlibs/testrunner/utils.py (hunk 2 above)
Comment (miri64) [reply dist/pythonlibs/testrunner/utils.py]: Ah, just saw that they are not exactly the same, but still:

```py
def _test_utils_interactive_sync(child, retries, delay, ready_cmd='r', ready_exp='READY'):
    """Synchronisation for 'test_utils_interactive_sync' function
    when shell module is used.
    Interacts through input to wait for node being ready.
    """
    for _ in range(0, retries):
        child.sendline(ready_cmd)
        ret = child.expect_exact([ready_exp, pexpect.TIMEOUT], timeout=delay)
        if ret == 0:
            break
    else:
        # Last call to make it fail here,
        child.expect_exact(ready_exp, timeout=0)


def test_utils_interactive_sync(child, retries, delay):
    _test_utils_interactive_sync(child, retries, delay)
    child.sendline('s')
    child.expect_exact('START')


def test_utils_interactive_sync_shell(child, retries, delay):
    _test_utils_interactive_sync(child, retries, delay, '\n', '>')
```

File: dist/pythonlibs/testrunner/utils.py [hunk 3]
```diff
@@ -9,19 +9,33 @@
...
     child.expect_exact('START')
+
+
+def test_utils_interactive_sync_shell(child, retries, delay):
+    """Synchronisation `shell` and `test_utils_interactive_sync` modules are
+    used ('test_utils_interactive_sync' function is not)
+
+    Interacts through input to wait for node being ready.
+    """
+    _test_utils_interactive_sync(child, retries, delay, '\n', '>')
```
Comment (miri64) [inline dist/pythonlibs/testrunner/utils.py]: Whitespace error here, according to the GitHub renderer.

File: dist/pythonlibs/testrunner/utils.py [hunk 4]
```diff
@@ -9,19 +9,34 @@
...
+
```
Comment (miri64) [inline dist/pythonlibs/testrunner/utils.py]: Still there... not sure what the problem is exactly. Do you use some (UNIX) exotic editor?

Comment (miri64) [reply dist/pythonlibs/testrunner/utils.py]: The git whitespace checker does not seem to have a problem with this, however `flake8` has https://travis-ci.org/github/RIOT-OS/RIOT/builds/661152542#L197

Comment (fjmolinas) [reply dist/pythonlibs/testrunner/utils.py]: nope, just bodged it :D

File: dist/pythonlibs/testrunner/utils.py (hunk 3 above)
Comment (fjmolinas) [inline dist/pythonlibs/testrunner/utils.py]: Pyterm doesn't do anything on an empty line... so this doesn't work o pyterm.

Comment (fjmolinas) [reply dist/pythonlibs/testrunner/utils.py]: I was using `term` isntead of `cleanterm` which is used in tests

**Comment by fjmolinas** [comment]
Rebased, fixed failing tests. I added a `test_utils_interactive_sync_shell` pseudomodule.

**Comment by fjmolinas** [comment]
@miri64 there are two failures but unrelated, could you give this a second look?

**Comment by miri64** [comment]
Are you sure https://ci.riot-os.org/RIOT-OS/RIOT/13613/f8c342645f36d85f6e0a2192b27c112c1f3d940f/output/run_test/tests/netstats_l2/esp32-wroom-32:gnu.txt is unrelated? With https://ci.riot-os.org/RIOT-OS/RIOT/13613/f8c342645f36d85f6e0a2192b27c112c1f3d940f/output/run_test/examples/suit_update/nrf52dk:gnu.txt I agree, however 

```
/bin/sh: 1: arm-none-eabi-gcc: not found
/tmp/dwq.0.6171555738823028/7c4b6d9e5c842d0fd6803d8d5f2d270d/dist/tools/jlink/jlink.sh flash /tmp/dwq.0.6171555738823028/7c4b6d9e5c842d0fd6803d8d5f2d270d/examples/suit_update/bin/nrf52dk/suit_update-slot0-extended.bin
```

This _should_ be easily fixable on the worker @kaspar030 or am I mistaking? ;-)

**Comment by fjmolinas** [comment]
> Are you sure https://ci.riot-os.org/RIOT-OS/RIOT/13613/f8c342645f36d85f6e0a2192b27c112c1f3d940f/output/run_test/tests/netstats_l2/esp32-wroom-32:gnu.txt is unrelated? With https://ci.riot-os.org/RIOT-OS/RIOT/13613/f8c342645f36d85f6e0a2192b27c112c1f3d940f/output/run_test/examples/suit_update/nrf52dk:gnu.txt I agree, however

That test fails often on nightlies and on regular tests. So to be honest, everytime I see it fail I don't look into it in detail. But I have a `esp32-wroom-32` remotely, I'll test

**Comment by fjmolinas** [comment]
> That test fails often on nightlies and on regular tests. So to be honest, everytime I see it fail I don't look into it in detail. But I have a esp32-wroom-32 remotely, I'll test

Hmm seems its un-plugged or has some kind of issue, I can't do anything about it for now... @aabadie do you have an `espr32-wroom-32`?

**Comment by miri64** [comment]
While we wait for confirmation, let's run this on Murdock one last time. Please squash

**Comment by fjmolinas** [comment]
> While we wait for confirmation, let's run this on Murdock one last time. Please squash

Confirmation came from murdock :)

File: Makefile.dep [hunk 5]
```diff
@@ -1102,5 +1108,11 @@ else
     USEMODULE += $(filter periph_init_%,$(filter-out $(DISABLE_MODULE),$(DEFAULT_MODULE)))
   endif
 
+  # Add test_utils_interactive_sync_shell
+  ifneq (,$(filter test_utils_interactive_sync,$(USEMODULE)))
+      USEMODULE += $(filter test_utils_interactive_sync_%, \
```
Comment (aabadie) [inline Makefile.dep]:
```suggestion
    USEMODULE += $(filter test_utils_interactive_sync_%, \
```

File: dist/pythonlibs/testrunner/utils.py [hunk 6]
```diff
@@ -9,19 +9,33 @@
...
             break
     else:
-        # Last call to make it fail her,
-        child.expect_exact('READY', timeout=0)
+        # Last call to make it fail here,
+        child.expect_exact(ready_exp, timeout=0)
+
 
+def test_utils_interactive_sync(child, retries, delay):
+    """Synchronisation for 'test_utils_interactive_sync' function
```
Comment (aabadie) [inline dist/pythonlibs/testrunner/utils.py]:
```suggestion
    """Synchronization for 'test_utils_interactive_sync' function
```

File: dist/pythonlibs/testrunner/utils.py [hunk 7]
```diff
@@ -9,19 +9,33 @@
...
+
+    Interacts through input to wait for node being ready.
+    """
+    _test_utils_interactive_sync(child, retries, delay)
     child.sendline('s')
     child.expect_exact('START')
+
+
+def test_utils_interactive_sync_shell(child, retries, delay):
+    """Synchronisation `shell` and `test_utils_interactive_sync` modules are
```
Comment (aabadie) [inline dist/pythonlibs/testrunner/utils.py]:
```suggestion
    """Synchronization `shell` and `test_utils_interactive_sync` modules are
```

File: tests/ps_schedstatistics/Makefile [hunk 8]
```diff
@@ -6,4 +6,9 @@ USEMODULE += ps
 USEMODULE += schedstatistics
 USEMODULE += printf_float
 
+# For this test we don't want to use the shell version of
+# test_utils_interactive_sync, sinc we want to synchronize before
```
Comment (aabadie) [inline tests/ps_schedstatistics/Makefile]: sinc ?

File: tests/test_tools/Makefile [hunk 9]
```diff
@@ -6,4 +6,8 @@ USEMODULE += shell
 # Disable shell echo and prompt to not have them in the way for testing
 CFLAGS += -DSHELL_NO_ECHO=1 -DSHELL_NO_PROMPT=1
 
+# No need for test_utils_interactive_sync in this test since the test
+# synchronizes by itself though `shellping` command.
```
Comment (aabadie) [inline tests/test_tools/Makefile]:
```suggestion
# synchronizes by itself through `shellping` command.
```

**Pull request merged**